scripts/build-engine/src/constant-manager/config-interface.ts

```diff
@@ -0,0 +1,34 @@
+export interface IConstantInfo {
+    /**
+     * The comment of the constant.
+     * Which is used to generate the consts.d.ts file.
+     */
+    readonly comment: string;
+    /**
+     * The default value of the constant.
+     * It can be a boolean or string.
+     * When it's a string type, the value is the result of eval().
+     */
+    value: boolean | string,
+    /**
+     * Whether exported to global as a `CC_XXXX` constant.
+     * eg. WECHAT is exported to global.CC_WECHAT
+     * NOTE: this is a feature of compatibility with Cocos 2.x engine.
+     */
+    readonly ccGlobal: boolean,
+    /**
+     * Whether exported to developer.
+     * If true, it's only exported to engine.
+     */
+    readonly internal: boolean,
+    /**
+     * Some constant can't specify the value in the Editor, Preview or Test environment,
+     * so we need to dynamically judge them in runtime.
+     * These values are specified in a helper called `helper-dynamic-constants.ts`.
+     */
+    readonly dynamic: boolean
```

Comment:
Given that fewer constants are dynamic, can we allow `dynamic` to be optional and default to be false? Similar, `ccGlobal` optional and default to be false since later added constants are not belong to legacy stuff. As for the internal, it should keep mandatory to avoid being acidently exposed in userland.

Reply:
updated